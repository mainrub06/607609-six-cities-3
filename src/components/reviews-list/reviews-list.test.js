import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list.jsx";

const REVIEWS = [
  {
    id: 1,
    rate: 2,
    comment: `review.comment`,
    date: `review.date`,
    user: {
      id: 1,
      isPro: false,
      name: `review.user.name`,
      avatar: `review.user.avatar_url`
    }
  },
  {
    id: 2,
    rate: 3,
    comment: `review.comment`,
    date: `review.date`,
    user: {
      id: 1,
      isPro: false,
      name: `review.user.name`,
      avatar: `review.user.avatar_url`
    }
  }
];

it(`Render ReviewsList component`, () => {
  const tree = renderer
    .create(
        <ReviewsList reviews = {REVIEWS}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
