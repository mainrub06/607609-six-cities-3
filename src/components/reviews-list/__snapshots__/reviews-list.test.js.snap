// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render snapshot <ReviewsList/> 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="review.user.name"
          className="reviews__avatar user__avatar"
          height="54"
          src="review.user.avatar_url"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        review.user.name
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 40,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        review.comment
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        Invalid Date
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="review.user.name"
          className="reviews__avatar user__avatar"
          height="54"
          src="review.user.avatar_url"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        review.user.name
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 60,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        review.comment
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        Invalid Date
      </time>
    </div>
  </li>
</ul>
`;
